assets/js/linear_algebra/dot_product_symmetric.js

```diff
@@ -316,40 +316,61 @@ function set_position(pos){
 
 
 function swap(u, v){
-  let uTv = lib.dot_product(u, v),
-      texts_for_table = texts_to_show(u, v),
-      uTv_text_table = texts_for_table[0],
-      vTu_text_table = texts_for_table[1];
-
-  let uTv_texts = [
-      {text:'u\u1d40v = '.concat(uTv.toFixed(3)),
-       x: (v.x * uTv/2).toFixed(2),
-       y: (v.y * uTv/2).toFixed(2),
-       text_color: 0, text_opacity: 1,
-       font_size: 15, tt: 0, key: 'uTv_texts'}
-  ];
+  let uTv = lib.dot_product(u, v);
 
-  lib.plot_texts(uTv_texts, tt=0, name='uTv_texts');
+  let [lines_vector_u, texts_vector_u] = lib.text_matrix_to_list(
+          [[u.coord.x.toFixed(2), u.coord.y.toFixed(2), u.coord.z.toFixed(2)]],
+          [last_col_coord, start_coord_y],
+          14, [['xu', 'yu', 'zu']], 5
+          ),
+      [lines_vector_v, texts_vector_v] = lib.text_matrix_to_list(
+          [[v.coord.x.toFixed(2), v.coord.y.toFixed(2), v.coord.z.toFixed(2)]],
+          [last_col_coord, start_coord_y],
+          14, [['xv', 'yv', 'zv']], 15
+          ),
+      [lines_uT, texts_uT] = lib.text_matrix_to_list(
+          [[u.coord.x.toFixed(2)], [u.coord.y.toFixed(2)], [u.coord.z.toFixed(2)]],
+          [start_coord_x, last_row_coord],
+          14, [['xu'], ['yu'], ['zu']], 5
+          ),
+      [lines_vT, texts_vT] = lib.text_matrix_to_list(
+          [[v.coord.x.toFixed(2)], [v.coord.y.toFixed(2)], [v.coord.z.toFixed(2)]],
+          [start_coord_x, last_row_coord],
+          14, [['xv'], ['yv'], ['zv']], 15
+          );
+
+  let uTv_texts = [],
+      uTv_lines = [],
+      vTu_texts = [],
+      vTu_lines = [];
+
+
+  uTv_texts.push(...texts_uT);
+  uTv_texts.push(...texts_vector_v);
+  uTv_texts.push(uT_name, v_name);
+
+  uTv_lines.push(...lines_uT);
+  uTv_lines.push(...lines_vector_v);
+
+  vTu_texts.push(...texts_vT);
+  vTu_texts.push(...texts_vector_u);
+  vTu_texts.push(vT_name, u_name);
+
+  vTu_lines.push(...lines_vT);
+  vTu_lines.push(...lines_vector_u);
+
+  console.log(uTv_lines, vTu_lines);
 
   if (position_state == 0) {
-    lib.plot_texts(lib.text_table_to_list(
-      vTu_text_table,
-      start_coord_x=start_coord_x, start_coord_y=start_coord_y,
-      w_unit=w_unit, h_unit=h_unit,
-      dws_array=dws_array, dhs_array=dhs_array),
-      tt=1000, name='transition');
+    lib.plot_texts(vTu_texts, tt, 'transition');
+    lib.plot_lines(vTu_lines, tt, 'bracket');
     position_state = 1;
-
+    
   } else {
-    lib.plot_texts(lib.text_table_to_list(
-      uTv_text_table,
-      start_coord_x=start_coord_x, start_coord_y=start_coord_y,
-      w_unit=w_unit, h_unit=h_unit,
-      dws_array=dws_array, dhs_array=dhs_array),
-    tt=1000, name='transition');
-    position_state = 0;
-  };
-
+      lib.plot_texts(uTv_texts, tt, 'transition');
+      lib.plot_lines(uTv_lines, tt, 'bracket');
+      position_state = 0;
```

Comment:
function `swap` doesn't have `tt` argument. this should be a constant, e.g. `1000`

Reply:
done